rules/custom-error.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+const findSuperExpression = node => node.type === 'ExpressionStatement' && node.expression.type === 'CallExpression' && node.expression.callee.type === 'Super';
+
+const findAssignmentExpression = (node, name) => {
+	if (node.type !== 'ExpressionStatement' || node.expression.type !== 'AssignmentExpression') {
+		return false;
+	}
+
+	const lhs = node.expression.left;
+
+	if (!lhs.object || lhs.object.type !== 'ThisExpression') {
+		return false;
+	}
+
+	return lhs.property.name === name;
+};
+
+const create = context => {
+	return {
+		ClassDeclaration: node => {
+			if (node.superClass && node.superClass.name === 'Error') {
+				const name = node.id.name;
+				const body = node.body.body;
+
+				const constructor = body.find(x => x.kind === 'constructor');
+
+				if (!constructor) {
+					context.report({
+						node,
+						message: 'Add a constructor to your error'
+					});
+					return;
+				}
+
+				const constructorBody = constructor.value.body.body;
+
+				const superExpression = constructorBody.find(findSuperExpression);
+
+				if (!superExpression) {
+					context.report({
+						node: constructor.value.body,
+						message: 'No call to `super()`'
+					});
+				} else if (superExpression.expression.arguments.length === 0) {
+					context.report({
+						node, // superExpression.expression -> Tests fail
```

Comment:
Maybe we should drop this check entirely, also the one you suggested. What if, not sure why though, you accept an argument in the constructor but aren't setting a message.

Just a stupid example which doesn't make much sense but for the sake of showing you.

```js
class HTTPError extends Error {
    constructor(statusCode) {
        super();
        this.statusCode = statusCode;
    }
}
```